### PR TITLE
Secure file permissions contextmanager

### LINKED ISSUE #1252: Unclosed file handle by get_file_descriptor_for_new_secure_file
## Describe the bug
`monkey/monkey_island/cc/server_utils/encryptor.py:30` leaves a file handle open. Either `get_file_descriptor_for_new_secure_file` should not return a handle and the method should be renamed or the handle should be used in the proceeding statement. 

## Machine version (please complete the following information):
 - OS: Windows and Linux

### PR DESCRIPTION
**Pull request by mssalvatore**
# What does this PR do? 

Fixes #1252

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing? 
* [ ] ~~Was the CHANGELOG.md updated to reflect the changes?~~
* [ ] ~~Was the documentation framework updated to reflect the changes?~~

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by running on Windows and running unit tests
* [ ] ~~If applicable, add screenshots or log transcripts of the feature working~~

### CODE REVIEW

File: monkey/monkey_island/cc/server_utils/file_utils.py [hunk 1]
```diff
@@ -79,16 +85,15 @@ def _get_file_descriptor_for_new_secure_file_linux(path: str) -> int:
 def _get_file_descriptor_for_new_secure_file_windows(path: str) -> int:
     try:
         file_access = win32file.GENERIC_READ | win32file.GENERIC_WRITE
-        # subsequent open operations on the object will succeed only if read access is requested
-        file_sharing = win32file.FILE_SHARE_READ
+        file_sharing = win32file.FILE_SHARE_READ | win32file.FILE_SHARE_WRITE
```
Comment (shreyamalviya) [inline monkey/monkey_island/cc/server_utils/file_utils.py:88]: Why this change?

Comment (mssalvatore) [reply monkey/monkey_island/cc/server_utils/file_utils.py:88]: From the documentation:

Comment (mssalvatore) [reply monkey/monkey_island/cc/server_utils/file_utils.py:88]: From the documentation:
> Enables subsequent open operations on a file or device to request write access.
> Otherwise, other processes cannot open the file or device if they request write access.

It's a bit ambiguous as to whether or not the current process can open for writing without `win32file.FILE_SHARE_WRITE`

**Review by shreyamalviya — APPROVED** [review]
